### PR TITLE
Replace Map<Integer, E> with ArrayList<E> where possible

### PR DESCRIPTION
**Pull request by HannesWell**
While looking into the `GnpRandomGraphGenerator` I noticed that it is using a Map<Integer, E> where the keys are consecutive int values. In this case a ArrayList is the better data structure to choose in terms of runtime and memory demand.

Getting an element at a certain index of a `ArrayList` is much faster than getting the value of a key from a `HashMap` plus autoboxing from int to integer objects is avoided. Additionally the memory demand of a ArrayList with the same number of elements/entrys is clearly smaller because the backing array is smaller and entry-Objects are avoided.
Furthermore Map<Integer,Integer> is replaced by an int-array, if possible.

Using a Integer-key in maps is only useful if the keys are not consecutive values.

There are many more places in JGraphT where Maps with Integer-keys are used but from a first look at the classes I cannot say if the integer-keys are consecutive. If you know more classes where this change is applicable, I can include them.

----

- [x] I read and understood <https://github.com/jgrapht/jgrapht/wiki/Become-a-Contributor>
- [x] I read and understood <https://github.com/jgrapht/jgrapht/wiki/How-to-make-your-first-%28code%29-contribution>
- [ ] I added [unit tests](https://github.com/jgrapht/jgrapht/wiki/Unit-testing)
- [ ] I added [documentation](https://github.com/jgrapht/jgrapht/wiki/How-to-write-documentation)
- [x] I followed the [Coding and Style Conventions](https://github.com/jgrapht/jgrapht/wiki/Coding-and-Style-Conventions)
- [x] I **have not** modified `HISTORY.md` or `CONTRIBUTORS.md`
- [x] I ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)

### CODE REVIEW

**Comment by d-michail** [comment]
Could you please respect the current formatting, as its gets very difficult to see changes if the formatting changes? We have a xml configuration for eclipse in the `etc/` folder.

**Comment by HannesWell** [comment]
Pardon me. It wasn't the formatter, I'm using the JGraphT formatter already.
But I have save-actions enabled in my Eclipse to clean up my code (e.g. enforce braces for all control structures) braces to and these kicked in here. Personally I really like them and they are quite powerful to enforce some more basic style conventions. But since they are not used in JGraphT these changes pollute the general changes.
Therefore I reverted these clean up actions.

File: jgrapht-core/src/main/java/org/jgrapht/generate/GridGraphGenerator.java
Comment (Toptachamann) [inline jgrapht-core/src/main/java/org/jgrapht/generate/GridGraphGenerator.java:74]: Preallocate memory in the array list by specifying its size. Also, consider renaming this variable to something link `vertices` or `vertexList` (the previous name `map` wasn't good either, so let's improve this).

File: jgrapht-io/src/main/java/org/jgrapht/nio/dimacs/DIMACSImporter.java
Comment (Toptachamann) [inline jgrapht-io/src/main/java/org/jgrapht/nio/dimacs/DIMACSImporter.java:162]: Consider initializing this variable in the constructor.

**Review by Toptachamann — CHANGES_REQUESTED** [review]
This PR looks good to me. I've left a couple of minor comments. Once they're addressed, this PR can be merged unless there're more comments.

File: jgrapht-core/src/main/java/org/jgrapht/generate/GridGraphGenerator.java
Comment (HannesWell) [reply jgrapht-core/src/main/java/org/jgrapht/generate/GridGraphGenerator.java:74]: Agree.
I changed the name to vertices

File: jgrapht-io/src/main/java/org/jgrapht/nio/dimacs/DIMACSImporter.java
Comment (HannesWell) [reply jgrapht-io/src/main/java/org/jgrapht/nio/dimacs/DIMACSImporter.java:162]: Alright.
Due to the design of the class (fields assigned with lambdas that refer to other fields) the field then cannot be final (like the graph field).

What do you think about, discarding the `Consumers` class and simplify creating the lambdas in two methods `createNodeCountConsumer(Graph<V,E>,List<V>)` and `createEdgeConsumer(Graph<V,E>,List<V>)` of the surrounding class?
This would be simpler than it is now. And because the parameter Graph and List are captured by the lambdas there is no need to model fields explicitly.

**Comment by HannesWell** [comment]
Thanks for the review. I applied the changes you requested.

**Review by Toptachamann — APPROVED** [review]

File: jgrapht-io/src/main/java/org/jgrapht/nio/dimacs/DIMACSImporter.java
Comment (Toptachamann) [reply jgrapht-io/src/main/java/org/jgrapht/nio/dimacs/DIMACSImporter.java]: These two consumers depend on the common vertex list. IIUC, while creating consumers in two separate methods, you'll still need to keep track of both graph and this common list in both lambdas. The advantage of the current code is that this dependency is very clear from the code, so I don't think it's worth it.

Comment (HannesWell) [reply jgrapht-io/src/main/java/org/jgrapht/nio/dimacs/DIMACSImporter.java]: Alright, if you prefer the current code, I'm fine with that and just leaf it as it is.

**Comment by Toptachamann** [comment]
Great job, thank you!

**Comment by HannesWell** [comment]
Thanks.

**Pull request merged**
